arraycontext/fake_numpy.py

```diff
@@ -61,6 +62,13 @@ def get(c_name, nargs, naxes):
                     ],
                 name="actx_special_%s" % c_name,
                 tags=(ElementwiseMapKernelTag(),))
+        for arg in prg.default_entrypoint.args:
+            if isinstance(arg, lp.ArrayArg):
+                arg.tags = [IsDOFArray()]
+            if arg.name =="out":
+                arg.is_output_only = True
```

Comment:
Never modify lopoy data structures in-place.

Reply:
Reworked this.